tests/conftest.py

```diff
@@ -85,17 +85,9 @@ def aioclient_mock():
 @pytest.fixture
 def mqtt_mock(loop, hass):
     """Fixture to mock MQTT."""
-    with patch('homeassistant.components.mqtt.MQTT') as mock_mqtt:
-        mock_mqtt().async_connect.return_value = mock_coro(True)
-        assert loop.run_until_complete(setup.async_setup_component(
-            hass, mqtt.DOMAIN, {
-                mqtt.DOMAIN: {
-                    mqtt.CONF_BROKER: 'mock-broker',
-                }
-            }))
-        client = mock_mqtt()
-        client.reset_mock()
-        return client
+    client = loop.run_until_complete(async_mock_mqtt_component(hass))()
+    client.reset_mock()
```

Comment:
I'm not sure why `reset_mock()` is used here - leaving it there for now...

Reply:
It was used here to reset `client.mock_calls` to be empty. Since initializing the component calls the client, it came with added mocks that would never be tested upon.

Reply:
Ah, that makes sense 👍